src/pymatgen/io/vasp/inputs.py

```diff
@@ -734,7 +735,8 @@ def __init__(self, params: dict[str, Any] | None = None) -> None:
                     val.append(params["MAGMOM"][idx * 3 : (idx + 1) * 3])
                 params["MAGMOM"] = val
 
-            self.update(params)
```

Comment:
I have to drop the `update` method to use the setter method, I don't think this would break anything as the dict is just init. But still good to get more eyes on this change.